main.go

```diff
@@ -126,73 +109,251 @@ func (e *periodicExporter) Collect(ch chan<- prometheus.Metric) {
-				float64(stat.Statistics.CpusUserTimeSecs),
-				stat.Source, host, stat.FrameworkId,
-			)
-			metricsChan <- prometheus.MustNewConstMetric(
-				memLimitDesc,
-				prometheus.GaugeValue,
-				float64(stat.Statistics.MemLimitBytes),
-				stat.Source, host, stat.FrameworkId,
-			)
-			metricsChan <- prometheus.MustNewConstMetric(
-				memRssDesc,
-				prometheus.GaugeValue,
-				float64(stat.Statistics.MemRssBytes),
-				stat.Source, host, stat.FrameworkId,
-			)
-		}
+func (e *periodicExporter) fetchNetworkStatistics(url *url.URL, host string, metricsChan chan <- prometheus.Metric) {
+	monitorURL := fmt.Sprintf("%s/monitor/statistics.json", url)
+	resp, err := httpClient.Get(monitorURL)
+	if err != nil {
+		log.Warn(err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	defer resp.Body.Close()
+
+	var stats []mesos_stats.Monitor
+	if err = json.NewDecoder(resp.Body).Decode(&stats); err != nil {
+		log.Warn("failed to deserialize response: ", err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	labels := []string{"task", "slave", "framework_id"}
+	for _, stat := range stats {
+		labelValues := []string{stat.Source, host, stat.FrameworkId}
+		e.gauge(metricsChan, labels, labelValues, "mesos_task_cpu_limit", float64(stat.Statistics.CpusLimit), "Fractional CPU limit.")
+		e.counter(metricsChan, labels, labelValues, "mesos_task_cpu_system_seconds_total", float64(stat.Statistics.CpusSystemTimeSecs), "Cumulative system CPU time in seconds.")
+		e.counter(metricsChan, labels, labelValues, "mesos_task_cpu_user_seconds_total", float64(stat.Statistics.CpusUserTimeSecs), "Cumulative system CPU time in seconds.")
+		e.gauge(metricsChan, labels, labelValues, "mesos_task_memory_limit_bytes", float64(stat.Statistics.MemLimitBytes), "Task memory limit in bytes.")
+		e.gauge(metricsChan, labels, labelValues, "mesos_task_memory_rss_bytes", float64(stat.Statistics.MemRssBytes), "Task memory RSS usage in bytes.")
 	}
 }
 
+func (e *periodicExporter) fetchSlaveMetricsSnapshot(url *url.URL, host string, metricsChan chan <- prometheus.Metric) {
+	monitorURL := fmt.Sprintf("%s/metrics/snapshot", url)
+	resp, err := httpClient.Get(monitorURL)
+	if err != nil {
+		log.Warn(err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	defer resp.Body.Close()
+
+	var stats mesos_metrics.SlaveMetrics
+	if err = json.NewDecoder(resp.Body).Decode(&stats); err != nil {
+		log.Warn("failed to deserialize response: ", err)
+		e.errors.WithLabelValues(host).Inc()
+		return
+	}
+	labels := []string{"node"}
+	labelValues := []string{host}
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_resource_cpus_available", float64(stats.CpuTotal), "Amount of CPUs available as resource")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_resource_cpus_used", stats.CpuUsed, "Amount of CPU resources in use")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_resource_cpus_used_percentage", stats.CpuPercentage, "Percentage of CPU resources in use")
```

Comment:
If you're going to export this it should be as a ratio, not a percentage. Having a mix of ratios and percentages in a system gets confusing fast.

As this can be calculated from the other two, I'd suggest dropping it. This also reduces the risk of someone taking an average of it.